packages/react/src/providers/FuelProvider.tsx

```diff
@@ -11,15 +13,22 @@ export { useConnectUI } from './FuelUIProvider';
 type FuelProviderProps = {
   ui?: boolean;
   fuelConfig?: FuelConfig;
+  /**
+   * Whether enforce connectors to be on the desired the network. Requires a FuelChainProvider wrapper.
+   * @default true
+   */
+  monitorNetwork?: boolean;
```

Comment:
I would change this config to receive:

- provider | providerUrl | number
- If DApp configures a provider request to be the same one on the wallet check by chainId only.

And include FuelChainProvider inside the FuelUIProvider no need to a new provider in this case.